cmd/snap-confine/mount-support.c

```diff
@@ -878,6 +903,41 @@ static bool __attribute__((used))
 	return false;
 }
 
+static struct sc_mount *sc_homedir_mounts(const struct sc_invocation *inv)
+{
+	if (inv->homedirs == NULL) {
+		// Return empty array, but never NULL as functions rely
+		// on the mounts array to be non-NULL
+		return calloc(1, sizeof(struct sc_mount));
+	}
+
+	int num_homedirs = 0;
+	for (char **path = inv->homedirs; *path != NULL; path++) {
+		num_homedirs++;
+	}
+
+	// We add one element for the end-of-array indicator.
+	struct sc_mount *mounts = calloc(num_homedirs + 1, sizeof(struct sc_mount));
+	if (mounts == NULL) {
+		die("cannot allocate mount data for homedirs");
+	}
+
+	struct sc_mount *current_mount = mounts;
+	for (char **path = inv->homedirs; *path != NULL; path++) {
+		debug("Adding homedir: %s", *path);
+		current_mount->path = *path;
```

Comment:
(super-duper-nitpick) I was wondering if we should stdup() here just to make reasoning who owns the memory easier but all of this is very local so I'm probably overthiking things [and again - rust to the rescue :)]

Reply:
I think this makes much better sense, the thought had struct me when I went through this code, never did it, so really happy you suggested this!